projectq/backends/_ibm/_ibm_http_client.py

```diff
@@ -21,75 +21,221 @@
+        r = super().request('POST', urljoin(_api_url, post_job_url), **argument)
+        r.raise_for_status()
+        r_json=r.json()
+        execution_id = r_json["id"]
+        return execution_id
+
+    def _get_result(self,device, execution_id, num_retries=3000,
+                    interval=1, verbose=False):
+        
+        job_status_url='Network/ibm-q/Groups/open/Projects/main/Jobs/'+execution_id
+        
+        if verbose:
+            print("Waiting for results. [Job ID: {}]".format(execution_id))
+
+        original_sigint_handler = signal.getsignal(signal.SIGINT)
 
+        def _handle_sigint_during_get_result(*_):
+            raise Exception("Interrupted. The ID of your submitted job is {}."
+                            .format(execution_id))
 
-_api_url = 'https://quantumexperience.ng.bluemix.net/api/'
+        try:
+            signal.signal(signal.SIGINT, _handle_sigint_during_get_result)
+
+            for retries in range(num_retries):
+                
+                argument={'allow_redirects': True, 'timeout': (self.timeout, None)}
+                r = super().request('GET', urljoin(_api_url, job_status_url), **argument)
+                r.raise_for_status()
+                r_json = r.json()
+                if r_json['status']=='COMPLETED':
+                    return r_json['qObjectResult']['results'][0]
+                elif r_json['status']!='RUNNING':
+                    raise Exception("Error while running the code: {}."
+                        .format(r_json['status']))
+                time.sleep(interval)
+                if self.is_online(device) and retries % 60 == 0:
+                    self.get_list_devices()
+                    if not self.is_online(device):
+                        raise DeviceOfflineError("Device went offline. The ID of "
+                                                "your submitted job is {}."
+                                                .format(execution_id))
+
+        finally:
+            if original_sigint_handler is not None:
+                signal.signal(signal.SIGINT, original_sigint_handler)
+
+        raise Exception("Timeout. The ID of your submitted job is {}."
+                        .format(execution_id))
 
+class DeviceTooSmall(Exception):
+    pass
 
 class DeviceOfflineError(Exception):
     pass
 
+def show_devices(token,verbose=False):    
+    """
+    Access the list of available devices and their properties (ex: for setup configuration)
 
-def is_online(device):
-    url = 'Backends/{}/queue/status'.format(device)
-    r = requests.get(urljoin(_api_url, url))
-    return r.json()['state']
+    Args:
+        token (str): IBM quantum experience user password.
```

Comment:
I would change "user password" to "user API token" to match what is displayed on the IBMQ website.